p2panda-sync/src/log_sync.rs

```diff
@@ -78,44 +76,41 @@ where
         let mut sink = into_cbor_sink(tx);
         let mut stream = into_cbor_stream(rx);
 
-        // Get the log ids which are associated with this topic.
+        // Get the log ids which are associated with this topic
         let Some(log_ids) = self.topic_map.get(topic).await else {
-            return Err(SyncError::Protocol("Unknown topic id".to_string()));
+            return Err(SyncError::Critical(format!("unknown {topic:?} topic")));
```

Comment:
If we tried to initate a sync session over a topic we didn't know, this smells like a bug